### ISSUE TITLE
Getting inconsistent results from 3D Forward Simulation of Gravity Anomaly

### ISSUE THREAD

**Issue by JAGR00P** [simpeg/simpeg#1594]
Dear Developer,

I imported a density data file (extracted from the model that I ran) to find the Gravity Anomaly. But the final results are almost same, no matter what type and scale of density distribution I use. Please help me resolve this issue. Thanks.

Below are the attached data files of  3d density distribution, and Gravity anomaly code in python.
[density_data.csv](https://github.com/user-attachments/files/18122988/density_data.csv)
[density2.csv](https://github.com/user-attachments/files/18123074/density2.csv)

Below is the Gravity anomaly code in python.
[Gravity_Anomaly.txt](https://github.com/user-attachments/files/18123109/Gravity_Anomaly.txt)

**Comment by JAGR00P** [comment]
Hi @santisoler. Thanks for the reply. 

My actual issue is not with exact values of Gravity Anomaly rather the distribution of it. My density distribution varies in x-direction, but I don't see any corresponding variation of Gravity anomaly in x-direction. The output is very symmetric across the centre point (which I dont expect). 

Also, I ran the code for two density files attached above, separately.

First I ran code by reading first file.
`density_data = pd.read_csv("density_data.csv")` 

After getting Gravity data of this, then I ran code by reading other file.
`density_data = pd.read_csv("density2.csv")` 

One file (density2.csv) contains dimensions of around 1200 x 1000 x 350 (in km) and other file (density_data.csv) contains very small portion extracted from previous file with dimensions around 150 x 100 x 100 (in km). 

Even though I get different values of Gravity Anomaly in both cases, but the distribution is almost the same (symmetric across center).  

My motive is not to compare both files but to see clearly the Gravity variation (plotted on x-y plane) above the subducted plate (visualised by taking slice in x-z plane of density data).

**Comment by santisoler** [comment]
Hi @JAGR00P. Thanks for the clarification, now I understand your situation better.

I don't think there's an issue with the forward modelling simulation, I think it's correctly reproducing the gravity effect of your model. You're computing the gravitational effect of a mesh whose cells have (almost) all positive density contrast. The observation points that are located around the center of the mesh are "surrounded" by positive density contrasts cells, feeling the effect of every one of them. The observation points that are close to the edge are also affected by every cell in the mesh, but they don't have any cell on the other side (outside the mesh), therefore the acceleration they feel is expected to be lower than the ones in the center.

I think the problem here arises from two reasons:
1. You're trying to model the gravity effect of every body in the subsurface. This means trying to reproduce the whole gravity acceleration. To do so, your model would need to include all bodies from Earth (core, mantle, all the crust, etc). This is not only non feasible, but also annoying. This is why we usually work with gravity disturbances (or even anomalies). Modelling gravity anomalies we only care about anomalous bodies that could have either positive or negative density contrasts.
2. Your mesh boundaries matches the boundaries of your observation points. This is ok if you assume that there's no anomalous body outside your survey area. But in order to run an inversion, for example, is highly recommended to add padding cells in order to account for those.

In summary: I'm not sure what your ultimate goal is, but if you are trying to forward model the subsurface with the total density, I'd consider to significantly expand my mesh to also consider masses outside the survey area (and also deeper sources as well). If you're trying to model the gravity anomaly, I'd convert those absolute density values to density contrast values.

Hope it helps!

**Comment by santisoler** [comment]
I'm closing this issue. Feel free to re-open it and comment on it if you think otherwise.
